speaker/layer2_controller.go

```diff
@@ -102,17 +102,18 @@ func (c *layer2Controller) ShouldAnnounce(l log.Logger, name string, toAnnounce
 	} else {
 		nodes = nodesWithActiveSpeakers(c.sList.UsableSpeakers())
 	}
-	ipString := toAnnounce.String()
-	// Sort the slice by the hash of node + load balancer ips. This
-	// produces an ordering of ready nodes that is unique to all the services
-	// with the same ip.
-	sort.Slice(nodes, func(i, j int) bool {
-		hi := sha256.Sum256([]byte(nodes[i] + "#" + ipString))
-		hj := sha256.Sum256([]byte(nodes[j] + "#" + ipString))
-
-		return bytes.Compare(hi[:], hj[:]) < 0
-	})
-
+	for _, toAnnounce := range lbIPs {
+		ipString := toAnnounce.String()
+		// Sort the slice by the hash of node + load balancer ips. This
+		// produces an ordering of ready nodes that is unique to all the services
+		// with the same ip.
+		sort.Slice(nodes, func(i, j int) bool {
```

Comment:
This will sort nodes twice which is really not necessary.
I'd always take the first ip and that's it.
What's important is to have a well defined rule to understand which speaker should announce the ip, so taking the first ip will work in both single & dual stack.